### DISCUSSION TITLE
Auto light/dark on MacOS with NvUI 3.0

### DISCUSSION THREAD

**Discussion by aronvandepol** [NvChad/NvChad#3046] [Show and tell]
I was trying to setup the auto-dark mode with the new NvUI 3.0 which required me to update my plugin config. 

figured I'd share it here for anyone looking for a similar setup. I haven't tested this on anything else than MacOS yet. 

i am using the melange theme that is included and a custom light version based on the light color variants in the original [melange-nvim theme repo](https://github.com/savq/melange-nvim/tree/master). 

The if statements check if we already have the correct theme set from the start to prevent unnecessary overwriting.

```lua
-- lua/plugins/auto-dark.lua
return {
  {
    "f-person/auto-dark-mode.nvim",
    event = "VeryLazy",
    config = function()
      local config = require("nvconfig") 
      require("auto-dark-mode").setup({
        update_interval = 1000,
        set_dark_mode = function()
          if config.base46.theme ~= "melange" then
            config.base46.theme = "melange"
            require("base46").load_all_highlights()
            print("Switched to dark theme")
          end
        end,
        set_light_mode = function()
          if config.base46.theme ~= "melange_light" then
            config.base46.theme = "melange_light"
            require("base46").load_all_highlights()
            print("Switched to light theme")
          end
        end,
      })

      require("auto-dark-mode").init()
    end,
  },
}

```

Edit: Removed .compile() based on the suggestion by @siduck

**Comment by siduck** [comment]
remove the .compile() because load_all_highlights() does that, compile just compiles, loadall.. will compile+refresh hlgroups

**Reply by aronvandepol** [reply]
Thank you! Adjusted the code :)

**Reply by siduck** [reply]
shouldnt you be using theme_toggle function?

![image](https://github.com/user-attachments/assets/07b53364-34d6-441f-ad46-f8812d70c3b5)

after that 

```lua
        set_dark_mode = function()
            require("base46").theme_toggle()
         end
         
```

same with light mode
